src/include/distributed/metadata_cache.h

```diff
@@ -116,7 +116,19 @@ typedef struct DistObjectCacheEntry
 	int colocationId;
 } DistObjectCacheEntry;
 
-
+extern bool IsNonDistributedTable(Oid relationId);
+extern bool IsReferenceTable(Oid relationId);
+extern bool IsDistributedTable(Oid relationId);
+extern bool IsAppendDistributedTable(Oid relationId);
+extern bool IsHashDistributedTable(Oid relationId);
+extern bool IsRangeDistributedTable(Oid relationId);
```

Comment:
Maybe `IsCitusTableType(oid, enum)` to avoid the word `Distributed`

```
HASH_DISTRIBUTED, APPEND_DISTRIBUTED, RANGE_DISTRIBUTED,
REFERENCE_TABLE, CITUS_LOCAL_TABLE, CITUS_TABLE_WITH_NO_DIST_KEY
```